dbt/models/marts/professional_development/dim_pl_engagement.sql

```diff
@@ -0,0 +1,82 @@
+with 
+
+pl_activity as ( 
+    select *
+    from {{ ref('dim_pl_activity') }}
+),
+
+school_years as ( 
+    select * 
+    from {{ ref('int_school_years') }}
+),
+
+active_teachers as ( 
+    select * 
+    from {{ ref('dim_teacher_status') }}
+    where status like 'active%'
+),
+
+active_teachers_sy_int as (
+    select 
+        active_teachers.*,
+        school_years.school_year_int 
+    from active_teachers 
+    join school_years 
+        on active_teachers.school_year = school_years.school_year 
+),
+
+pl_with_engagement as (
+    select 
+        pl_activity.teacher_id,
+        pl_activity.school_year,
+        school_years.school_year_int,
+        pl_activity.grade_band,
+        pl_activity.school_id,
+        pl_activity.school_district_id,
+        sum(pl_activity.num_hours) as total_hours,
+        sum(pl_activity.num_levels) as total_levels,
```

Comment:
```suggestion
        sum(coalesce(pl_activity.num_levels,0)) as total_levels,
```
Nulls can create issues when aggregating numerical values, so it might be better to set `total_hours` and ‘total_levels` to 0  when null